compose/bin/check-dependencies

```diff
@@ -0,0 +1,54 @@
+common_dependencies["2.4.4-p5"]="Composer:2.2 Elasticsearch:7.17 OpenSearch:1.2 MariaDB:10.4 MySQL:8.0 PHP:8.1 RabbitMQ:3.9 Redis:6.2 Varnish:7.3 Apache:2.4 nginx:1.22 'AWS-Aurora-(MySQL)':8.0 'AWS-S3':✔️ 'AWS-MQ':3.9.13 'AWS-ElastiCache':Redis6 'AWS-Elasticsearch':-- 'AWS-OpenSearch':1.2"
+common_dependencies["2.4.4-p4"]="Composer:2.2 Elasticsearch:7.17 OpenSearch:1.2 MariaDB:10.4 MySQL:8.0 PHP:8.1 RabbitMQ:3.9 Redis:6.2 Varnish:7.3 Apache:2.4 nginx:1.22 'AWS-Aurora-(MySQL)':8.0 'AWS-S3':✔️ 'AWS-MQ':3.9.13 'AWS-ElastiCache':Redis6 'AWS-Elasticsearch':-- 'AWS-OpenSearch':1.2"
+common_dependencies["2.4.4-p3"]="Composer:2.2 Elasticsearch:7.17 OpenSearch:1.2 MariaDB:10.4 MySQL:8.0 PHP:8.1 RabbitMQ:3.9 Redis:6.2 Varnish:7.1 Apache:2.4 nginx:1.22 'AWS-Aurora-(MySQL)':8.0 'AWS-S3':✔️ 'AWS-MQ':3.9.13 'AWS-ElastiCache':Redis6 'AWS-Elasticsearch':-- 'AWS-OpenSearch':1.2"
+common_dependencies["2.4.4-p2"]="Composer:2.1 Elasticsearch:7.16 OpenSearch:1.2 MariaDB:10.4 MySQL:8.0 PHP:8.1 RabbitMQ:3.9 Redis:6.2 Varnish:7.0 Apache:2.4 nginx:1.22 'AWS-Aurora-(MySQL)':5.7 'AWS-S3':✔️ 'AWS-MQ':3.8.11 'AWS-ElastiCache':Redis6 'AWS-Elasticsearch':7.9 'AWS-OpenSearch':--"
+common_dependencies["2.4.4-p1"]="Composer:2.1 Elasticsearch:7.16 OpenSearch:1.2 MariaDB:10.4 MySQL:8.0 PHP:8.1 RabbitMQ:3.9 Redis:6.2 Varnish:7.0 Apache:2.4 nginx:1.22 'AWS-Aurora-(MySQL)':5.7 'AWS-S3':✔️ 'AWS-MQ':3.8.11 'AWS-ElastiCache':Redis6 'AWS-Elasticsearch':7.9 'AWS-OpenSearch':--"
+common_dependencies["2.4.4"]="Composer:2.1 Elasticsearch:7.16 OpenSearch:1.2 MariaDB:10.4 MySQL:8.0 PHP:8.1 RabbitMQ:3.9 Redis:6.2 Varnish:7.0 Apache:2.4 nginx:1.22 'AWS-Aurora-(MySQL)':5.7 'AWS-S3':✔️ 'AWS-MQ':3.8.11 'AWS-ElastiCache':Redis6 'AWS-Elasticsearch':7.9 'AWS-OpenSearch':--"
+common_dependencies["2.4.3-p3"]="Composer:1 Elasticsearch:7.16 OpenSearch:1.2 MariaDB:10.4 MySQL:8.0 PHP:7.4 RabbitMQ:3.8 Redis:6.0 Varnish:6.5 Apache:2.4 nginx:1.18 'AWS-Aurora-(MySQL)':5.7 'AWS-S3':✔️ 'AWS-MQ':3.8.11 'AWS-ElastiCache':Redis6 'AWS-Elasticsearch':7.9 'AWS-OpenSearch':--"
+common_dependencies["2.4.2-p2"]="Composer:1 Elasticsearch:7.9 OpenSearch:-- MariaDB:10.4 MySQL:8.0 PHP:7.4 RabbitMQ:3.8 Redis:6.0 Varnish:6.4 Apache:2.4 nginx:1.18 'AWS-Aurora-(MySQL)':-- 'AWS-S3':✔️ 'AWS-MQ':-- 'AWS-ElastiCache':-- 'AWS-Elasticsearch':-- 'AWS-OpenSearch':--"
+
+print_magento_versions() {
+    echo "Available Magento 2 versions:"
+    for version in "${magento_versions[@]}"; do
+        echo "$version"
+    done
+}
+
+print_dependencies() {
+    local version=$1
+    IFS=' ' read -ra dependencies <<< "${common_dependencies[$version]}"
+    for dependency in "${dependencies[@]}"; do
+        echo "  • $dependency"
+    done
+}
+
+read -p "Enter the Magento 2 version you want to install: " magento_version
```

Comment:
This line fails shellcheck validation with:

read without -r will mangle backslashes.
See SC2162.

Per https://github.com/koalaman/shellcheck/wiki/SC2162

Reply:
I think we should also rename this from "Enter the Magento 2 version you want to install: " to "Enter the Magento 2 version you'd like to check dependencies on: ", as the former implies that we're about to install Magento, which we are not.

Reply:
Thx, I fixed it!

Reply:
> I think we should also rename this from "Enter the Magento 2 version you want to install: " to "Enter the Magento 2 version you'd like to check dependencies on: ", as the former implies that we're about to install Magento, which we are not.

This script was supposed to run before installing Magento, so maybe you misunderstood, this script shows the dependencies, not installing Magento. I changed it.

Reply:
The verbiage in this line "Enter the Magento 2 version you want to install:" implies that the user wants to install Magento. I understand this script does not do that, and just checks compatibility. For this reason, we need to update this to "Enter the Magento 2 version you'd like to check dependencies for:"